common/file_model/variant_allele.py

```diff
@@ -40,86 +42,64 @@ def get_prediction_results(self):
-                                                                                                                                               csq_record_list[consequence_index],
-                                                                                                                                               csq_record_list[sift_index], 
-                                                                                                                                               csq_record_list[polyphen_index]
-                                                                                                                                   ))
-                current_prediction_results = info_map[csq_record_list[allele_index]]["prediction_results"]
-                info_map[csq_record_list[allele_index]]["prediction_results"] += self.create_allele_prediction_results(current_prediction_results, 
-                                                                                                                                               csq_record_list[spdi_index], 
-                                                                                                                                               csq_record_list[cadd_index],
-                                                                                                                                               csq_record_list[gerp_index]
-                                                                                                                                               )
+            phenotypes = csq_record_list[prediction_index_map["phenotypes"]].split("&") if "phenotypes" in prediction_index_map.keys() else []
+            if csq_record_list[prediction_index_map["allele"]] not in info_map.keys():
+                info_map[csq_record_list[prediction_index_map["allele"]]] = {"phenotype_assertions": [], "predicted_molecular_consequences": [], "prediction_results": []} 
                 
+            for phenotype in phenotypes:
+                phenotype_assertions = self.create_allele_phenotype_assertion(phenotype) if phenotype else []
+                if (phenotype_assertions):
+                    info_map[csq_record_list[prediction_index_map["allele"]]]["phenotype_assertions"].append(phenotype_assertions)
+            predicted_molecular_consequences = self.create_allele_predicted_molecular_consequence(csq_record_list, prediction_index_map)
+            if (predicted_molecular_consequences):
+                info_map[csq_record_list[prediction_index_map["allele"]]]["predicted_molecular_consequences"].append(predicted_molecular_consequences)
+            if (info_map[csq_record_list[prediction_index_map["allele"]]]["prediction_results"]):
+                info_map[csq_record_list[prediction_index_map["allele"]]]["prediction_results"] += self.create_allele_prediction_results(prediction_results, csq_record_list, prediction_index_map)
```

Comment:
```suggestion
            current_prediction_results = info_map[csq_record_list[prediction_index_map["allele"]]]["prediction_results"]
            info_map[csq_record_list[prediction_index_map["allele"]]]["prediction_results"] += self.create_allele_prediction_results(current_prediction_results, csq_record_list, prediction_index_map)
```

and delete the `else` . See if it makes sense.

Reply:
I think it makes sense now, corrected this